### ISSUE TITLE
cleanup docs post GCS infra migration

### ISSUE THREAD

**Issue by neolit123** [kubernetes/kubeadm#2876]
see this and linked tracked issues:
- https://github.com/kubernetes/kubeadm/pull/2872
- https://github.com/kubernetes/kubeadm/pull/2872#issuecomment-1547866184
- [x] kubernetes/k8s.io#2396 (when this is closed we can update k/kubeadm)

the k/kubeadm docs contain some outdated examples. the PR above fixes some of them, but there is remaining work.

- [x] remove / change all gs:// entries in docs
- [x] remove /cluster/* script examples in docs
https://github.com/kubernetes/kubeadm/pull/2878

note we cannot change `gs://k8s-release-dev/ci/`
yet:
https://github.com/kubernetes/kubeadm/pull/2872#discussion_r1193581291

so this issue tracks future work.
but the /cluster change can be PRed now.

- https://github.com/kubernetes/kubeadm/pull/3148

**Comment by chendave** [comment]
Just a record, here is still reference to gcs bucket, https://github.com/kubernetes/kubeadm/blob/3d02f6cd3ac64a9446bece239588291ec12cdad2/docs/testing-pre-releases.md?plain=1#L117-L119

Is there any alternative to replace it? https://console.cloud.google.com/storage/browser/kubernetes-release/release/


When we have any other options, we can update the link above.

**Comment by chendave** [comment]
> remove / change all gs:// entries in docs

@neolit123  what do you mean for this todo item?  I searched in the code base, seems this has been addressed unless the `gs://k8s-release-dev/ci/` (I understood this will stay for an while).

**Comment by neolit123** [comment]
> Is there any alternative to replace it?

@rjsadow do you know if we can list these labels with curl?

**Comment by neolit123** [comment]
> > remove / change all gs:// entries in docs
> 
> @neolit123 what do you mean for this todo item? I searched in the code base, seems this has been addressed unless the `gs://k8s-release-dev/ci/` (I understood this will stay for an while).

we can't change it for now, afaik. it was discussed earlier as planned.

**Comment by rjsadow** [comment]
I don't believe there's a direct 1-for-1 replacement with curl in this case since directory browsing isn't available through the proxy. It might be worth reaching out to sig-release and seeing if we could add an index file or some other mechanism into the release bucket that provides a list of all latest and stable versions.

**Comment by rjsadow** [comment]
Let me take a crack at rewording it. The problem with removing it and referencing the release page is that only includes GA versions. If someone wants to test with a pre-release version they really should reference these files to get the tags to use. 

i.e.
```
❯ curl -L https://dl.k8s.io/release/latest-1.txt
v1.28.0-alpha.2
```
Which is the tag that should be used for the images. I don't know any better way to point someone to pre-release image tags.

**Comment by neolit123** [comment]
> Which is the tag that should be used for the images. I don't know any better way to point someone to pre-release image tags.

i don't know either for pre built ci images. in kubeadm e2e tests we build images from tars for a particular k/k commit. those tars are for sure available in the current ci bucket.
